bundles/persistence/org.openhab.persistence.h2sql/java/org/openhab/persistence/h2sql/internal/H2SqlPersistenceService.java

```diff
@@ -0,0 +1,464 @@
+        String value;
+        if (item instanceof DimmerItem || item instanceof RollershutterItem) {
+            value = item.getStateAs(PercentType.class).toString();
+            logger.debug("Got as Percent: {}", value);
+        } else if (item instanceof ColorItem) {
+            value = item.getStateAs(HSBType.class).toString();
+            logger.debug("Got as HSB: {}", value);
+        } else {
+            // All other items should return the best format by default
+            value = item.getState().toString();
+        }
+        logger.debug("H2SQL: State is {}::{}", item.getState(), value);
+
+        try {
+            statement = connection.createStatement();
+            sqlCmd = new String("INSERT INTO openhab." + item.getName() + " (TIME, VALUE) VALUES(NOW(),'" + value + "');");
+            statement.executeUpdate(sqlCmd);
+
+            long timerStop = System.currentTimeMillis();
+            logger.debug("H2SQL: Stored item '{}' as '{}'[{}] in {}ms", item.getName(), value,
+                    item.getState().toString(), timerStop - timerStart);
+            logger.debug("H2SQL: {}", sqlCmd);
+        } catch (Exception e) {
+            logger.error("H2SQL: Could not store item '{}' in database with statement '{}'", item.getName(), sqlCmd);
+            logger.error("     : " + e.getMessage());
+        } finally {
+            if (statement != null) {
+                try {
+                    statement.close();
+                } catch (Exception hidden) {
+                }
+            }
+        }
+    }
+
+    /**
+     * @{inheritDoc
+     */
+    public void store(Item item, String alias) {
+        store(item);
+    }
+
+    /**
+     * Checks if we have a database connection
+     * 
+     * @return true if connection has been established, false otherwise
+     */
+    private boolean isConnected() {
+        // Check if connection is valid
+        try {
+            if (connection != null && !connection.isValid(5000)) {
+                logger.error("H2SQL: Connection is not valid!");
+            }
+        } catch (SQLException e) {
+            logger.error("H2SQL: Error while checking connection");
+            logger.error("     : " + e.getMessage());
+        }
+        return connection != null;
+    }
+
+    /**
+     * Connects to the database
+     */
+    private void connectToDatabase() {
+        try {
+            logger.debug("H2SQL: Connecting to database");
+            Class.forName(driverClass).newInstance();
+
+            final String USERDATA_DIR_PROG_ARGUMENT = "smarthome.userdata";
+            final String eshUserDataFolder = System.getProperty(USERDATA_DIR_PROG_ARGUMENT);
+            String databaseFileName = "etc/";
+            if (eshUserDataFolder != null) {
+                databaseFileName = eshUserDataFolder + "/";
+            }
+            databaseFileName += "h2sql/openhab";
+
+            String url = "jdbc:h2:file:" + databaseFileName;
```

Comment:
all these declarations are static and should be declared at the top of this file